test/csuite/wt9199_checkpoint_txn_commit_race/main.c

```diff
@@ -0,0 +1,279 @@
+/*-
+ * Public Domain 2014-present MongoDB, Inc.
+ * Public Domain 2008-2014 WiredTiger, Inc.
+ *
+ * This is free and unencumbered software released into the public domain.
+ *
+ * Anyone is free to copy, modify, publish, use, compile, sell, or
+ * distribute this software, either in source code form or as a compiled
+ * binary, for any purpose, commercial or non-commercial, and by any
+ * means.
+ *
+ * In jurisdictions that recognize copyright laws, the author or authors
+ * of this software dedicate any and all copyright interest in the
+ * software to the public domain. We make this dedication for the benefit
+ * of the public at large and to the detriment of our heirs and
+ * successors. We intend this dedication to be an overt act of
+ * relinquishment in perpetuity of all present and future rights to this
+ * software under copyright law.
+ *
+ * THE SOFTWARE IS PROVIDED "AS IS", WITHOUT WARRANTY OF ANY KIND,
+ * EXPRESS OR IMPLIED, INCLUDING BUT NOT LIMITED TO THE WARRANTIES OF
+ * MERCHANTABILITY, FITNESS FOR A PARTICULAR PURPOSE AND NONINFRINGEMENT.
+ * IN NO EVENT SHALL THE AUTHORS BE LIABLE FOR ANY CLAIM, DAMAGES OR
+ * OTHER LIABILITY, WHETHER IN AN ACTION OF CONTRACT, TORT OR OTHERWISE,
+ * ARISING FROM, OUT OF OR IN CONNECTION WITH THE SOFTWARE OR THE USE OR
+ * OTHER DEALINGS IN THE SOFTWARE.
+ */
+#include "test_util.h"
+
+/*
+ * This test is to create a window where a checkpoint can fail to include a transaction's updates
+ * with commit times before stable. Consider a situation like below.
+ * 1. In the commit transaction the commit timestamp is checked and the transaction sleeps.
+ * 2. Stable timestamp is then moved past the commit timestamp.
+ * 3. Checkpoint selects the stable timestamp and runs. This checkpoint will not include changes
+ *    for a commit timestamp that is before the checkpoint's stable timestamp.
+ *
+ * Added a WT_TIMING_STRESS_COMMIT_TRANSACTION_SLOW timing stress to add a delay
+ * while committing a transaction.
+ */
+
+#define NUM_RECORDS WT_THOUSAND
+
+/* Constants and variables declaration. */
+
+static const char conn_config[] =
```

Comment:
I would prefer to write a continuously running stress test to verify that a checkpoint always include all the data it should include. However, this can be a separate ticket.